### PR TITLE
cpu: Add the microcode version to the CPU device

### PR DESCRIPTION
**Pull request by superm1**
I swear this used to do it at one point, but I noticed it was missing recently.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

**Comment by hughsie** [comment]
Shouldn't this be done using CPUID? I think mixing and matching sysfs-provided and cpuid-provided is probably a recipe for confusion.

**Comment by superm1** [comment]
> Shouldn't this be done using CPUID? I think mixing and matching sysfs-provided and cpuid-provided is probably a recipe for confusion.

Yeah, I double checked how it's supposed to be working.  The kernel issues a CPUID of 0x1 then reads the MSR.  I think we should be doing the same as a precaution.

Also it seems the kernel doesn't fetch 4 bytes in, but right afterwards.  The fixup I pushed seems to work.

**Comment by hughsie** [comment]
In main I get:

    Current version:    0x000000f0

with this branch I get nothing on my Lenovo Intel CPU :(

File: plugins/msr/fu-plugin-msr.c [hunk 1]
```diff
@@ -273,7 +276,7 @@ fu_plugin_msr_backend_device_added(FuPlugin *plugin, FuDevice *device, GError **
 		fu_dump_raw(G_LOG_DOMAIN, "IA32_BIOS_SIGN_ID", buf, sizeof(buf));
 		if (!fu_memread_uint32_safe(buf,
 					    sizeof(buf),
-					    0x4,
+					    0x0,
```
Comment (hughsie) [inline plugins/msr/fu-plugin-msr.c:279]: ya, i need 0x0 on INTEL...

Comment (hughsie) [reply plugins/msr/fu-plugin-msr.c:279]: On AMD is IA32_BIOS_SIGN_ID even valid?

Comment (superm1) [reply plugins/msr/fu-plugin-msr.c:279]: Hmm.  It's the same MSR for both AMD and Intel (0x8b).
* AMD code in the kernel: https://github.com/torvalds/linux/blob/master/arch/x86/kernel/cpu/amd.c#L610
* Intel code in the kernel: https://github.com/torvalds/linux/blob/master/arch/x86/include/asm/microcode_intel.h#L56.

But yeah it looks like in the kernel it's off by 4 for Intel and AMD.  I'll push a fixup.

**Review by hughsie — APPROVED** [review]
Thanks!

**Pull request merged**